code/game/machinery/embedded_controller/airlock_docking_controller_multi.dm

```diff
@@ -44,24 +39,21 @@
 		ui.set_auto_update(1)
 
 /obj/machinery/embedded_controller/radio/docking_port_multi/Topic(href, href_list)
-	return
+	return 1 // Apparently we swallow all input (this is corrected legacy code)
 
 
 
 //a docking port based on an airlock
+// This is the actual controller that will be commanded by the master defined above
 /obj/machinery/embedded_controller/radio/airlock/docking_port_multi
 	name = "docking port controller"
+	program = /datum/computer/file/embedded_program/airlock/multi_docking
 	var/master_tag	//for mapping
-	var/datum/computer/file/embedded_program/airlock/multi_docking/airlock_program
 	tag_secure = 1
 
-/obj/machinery/embedded_controller/radio/airlock/docking_port_multi/Initialize()
-	. = ..()
-	airlock_program = new/datum/computer/file/embedded_program/airlock/multi_docking(src)
-	program = airlock_program
-
 /obj/machinery/embedded_controller/radio/airlock/docking_port_multi/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
 	var/data[0]
+	var/datum/computer/file/embedded_program/airlock/multi_docking/airlock_program  // Cast to proper type
 
 	data = list(
 		"chamber_pressure" = round(airlock_program.memory["chamber_sensor_pressure"]),
```

Comment:
It is in fact not initialized!  Not noticed because bug doesn't affect docking, only viewing the UI, and this type of controller is rarely used.  Fixed now tho.